src/mixins/people-roles-mixin.ts

```diff
@@ -0,0 +1,203 @@
+    if (!rule) return true
+    return rule.test(this.completingParties.length)
+  }
+
+  /** Whether the Number of Incorporators rule is valid. Always true if rule doesn't exist. */
+  get validMinimumIncorporators (): boolean {
+    const rule = this.getPeopleAndRolesResource.rules.find(r => r.id === RuleIds.NUM_INCORPORATORS)
+    if (!rule) return true
+    return rule.test(this.incorporators.length)
+  }
+
+  /** Whether the Number of Directors rule is valid. Always true if rule doesn't exist. */
+  get validMinimumDirectors (): boolean {
+    const rule = this.getPeopleAndRolesResource.rules.find(r => r.id === RuleIds.NUM_DIRECTORS)
+    if (!rule) return true
+    return rule.test(this.directors.length)
+  }
+
+  /** Whether the Director Country rule is valid. Always true if rule doesn't exist. */
+  get validDirectorCountry (): boolean {
+    const rule = this.getPeopleAndRolesResource.rules.find(r => r.id === RuleIds.DIRECTOR_COUNTRY)
+    if (!rule) return true
+    const num = this.directors.filter(d => rule.test(d.mailingAddress.addressCountry)).length
+    // evaluate this rule only when there are enough minimum directors
+    return (this.validMinimumDirectors && num > this.directors.length / 2) // more than half
+  }
+
+  /** Whether the Director Province rule is valid. Always true if rule doesn't exist. */
+  get validDirectorProvince (): boolean {
+    const rule = this.getPeopleAndRolesResource.rules.find(r => r.id === RuleIds.DIRECTOR_PROVINCE)
+    if (!rule) return true
+    const num = this.directors.filter(
+      d => rule.test(d.mailingAddress.addressCountry, d.mailingAddress.addressRegion)
+    ).length
+    return (num > 0) // at least one
+  }
+
+  /** Whether the Number of Proprietor rule is valid. Always true if rule doesn't exist. */
+  get validNumProprietor (): boolean {
+    const rule = this.getPeopleAndRolesResource.rules.find(r => r.id === RuleIds.NUM_PROPRIETOR)
+    if (!rule) return true
+    return rule.test(this.proprietors.length)
+  }
+
+  /** Whether the Number of Partners rule is valid. Always true if rule doesn't exist. */
+  get validNumPartners (): boolean {
+    const rule = this.getPeopleAndRolesResource.rules.find(r => r.id === RuleIds.NUM_PARTNERS)
+    if (!rule) return true
+    return rule.test(this.partners.length)
+  }
+
+  /** Whether there are any people without roles. Used as a safety check. */
+  get validPeopleWithNoRoles (): boolean {
+    return (this.peopleWithNoRoles.length > 0)
+  }
+
+  /** Returns true if specified org/person is a director. */
+  isDirector (orgPerson: OrgPersonIF): boolean {
+    return orgPerson.roles.some(role => role.roleType === RoleTypes.DIRECTOR)
+  }
+
+  /** Helper to show the confirm dialog. */
+  private async showConfirmDialog (title: string, message: string, action: string): Promise<boolean> {
+    const confirm = await this.$refs.confirmDialog.open(title, message, {
```

Comment:
A little helper here that expects `confirmDialog` to exist on "this". Since this mixin is added to the PeopleAndRoles component's context (its "this"), this works.

TLDR; we can call the confirmDialog here but it needs to be declared in the component.